inc/renderer.php

```diff
@@ -133,101 +137,105 @@ public function generate_css( $post_id, $panels_data = false, $layout_data = fal
 				! empty( $row['style']['bottom_margin'] ) ||
 				! empty( $panels_margin_bottom_last_row )
 			) {
-				// Filter the bottom margin for this row with the arguments
 				$css->add_row_css( $post_id, $ri, '', array(
-					'margin-bottom' => apply_filters( 'siteorigin_panels_css_row_margin_bottom', $panels_margin_bottom . 'px', $row, $ri, $panels_data, $post_id )
+					'margin-bottom' => $panels_margin_bottom
 				) );
 			}
 
 			$collapse_order = ! empty( $row['style']['collapse_order'] ) ? $row['style']['collapse_order'] : ( ! is_rtl() ? 'left-top' : 'right-top' );
 
-			if ( $settings['responsive'] ) {
-
+			if ( $settings['responsive'] && ! empty( empty( $row['style']['collapse_behaviour'] ) ) ) {
 				// The default collapse behaviour
-				if ( empty( $row['style']['collapse_behaviour'] ) ) {
-
-					if (
-						$settings['tablet-layout'] &&
-						$cell_count >= 3 &&
-						$panels_tablet_width > $panels_mobile_width
-					) {
-						// Tablet responsive css for the row
-
-						$css->add_row_css( $post_id, $ri, array(
-							'.panel-no-style',
-							'.panel-has-style > .panel-row-style'
-						), array(
-							'-ms-flex-wrap'     => $collapse_order == 'left-top' ? 'wrap' : 'wrap-reverse',
-							'-webkit-flex-wrap' => $collapse_order == 'left-top' ? 'wrap' : 'wrap-reverse',
-							'flex-wrap'         => $collapse_order == 'left-top' ? 'wrap' : 'wrap-reverse',
-						), $panels_tablet_width . ':' . ( $panels_mobile_width + 1 ) );
-
-						$css->add_cell_css( $post_id, $ri, false, '', array(
-							'-ms-flex'      => '0 1 50%',
-							'-webkit-flex'  => '0 1 50%',
-							'flex'          => '0 1 50%',
-							'margin-right'  => '0',
-							'margin-bottom' => $panels_margin_bottom . 'px',
-						), $panels_tablet_width . ':' . ( $panels_mobile_width + 1 ) );
+				if (
+					$settings['tablet-layout'] &&
+					$cell_count >= 3 &&
+					$panels_tablet_width > $panels_mobile_width
+				) {
+					// Tablet responsive css for the row
 
+					$css->add_row_css( $post_id, $ri, array(
+						'.panel-no-style',
+						'.panel-has-style > .panel-row-style'
+					), array(
+						'-ms-flex-wrap'     => $collapse_order == 'left-top' ? 'wrap' : 'wrap-reverse',
+						'-webkit-flex-wrap' => $collapse_order == 'left-top' ? 'wrap' : 'wrap-reverse',
+						'flex-wrap'         => $collapse_order == 'left-top' ? 'wrap' : 'wrap-reverse',
+					), $panels_tablet_width . ':' . ( $panels_mobile_width + 1 ) );
 
-						$remove_bottom_margin = ':nth-';
-						if ( $collapse_order == 'left-top' ) {
-							$remove_bottom_margin .= 'last-child(' . ( count( $row['cells'] ) % 2 == 0 ? '-n+2' : '1' ) . ')';
-						} else {
-							$remove_bottom_margin .= 'child(-n+2)';
-						}
+					$css->add_cell_css( $post_id, $ri, false, '', array(
+						'-ms-flex'      => '0 1 50%',
+						'-webkit-flex'  => '0 1 50%',
+						'flex'          => '0 1 50%',
+						'margin-right'  => '0',
+						'margin-bottom' => $panels_margin_bottom,
```

Comment:
As far as I can tell, this is the only line that changed to remove the `'px` unit. That correct? Not sure why the diff marked so much as changed around this line. 😕